core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -554,6 +555,32 @@ public String authority() {
     }
   }
 
+  private final class DirectFallbackExecutor implements Executor {
```

Comment:
This is quite an interesting idea, although coupled with the ClientCallImpl change, it will cause us to use the `executor` after shutdown _more_ than previously. Previously we would _never_ use the `executor` after shutdown; only the deadlineCancellationExecutor.